tfhe/src/shortint/engine/server_side/mod.rs

```diff
@@ -159,13 +160,19 @@ impl ShortintEngine {
             &mut self.encryption_generator,
         );
 
+        let max_noise_level = MaxNoiseLevel::new(divide_ceil(
+            cks.parameters.carry_modulus().0 * cks.parameters.message_modulus().0 - 1,
+            cks.parameters.message_modulus().0 - 1,
+        ));
```

Comment:
i would probably use a plain divide (which floors) so that we are sure to respect the noise ? any reason to prefer the ceil ? (I believe all our current parameter sets should have round divisions for the noise)

Reply:
Indeed, replaced by a integer divide